statsmodels/multivariate/multivariate_ols.py

```diff
@@ -0,0 +1,472 @@
+    """
+    General linear model
+
+
+    Parameters
+    ----------
+    endog : array-like
+        Dependent variables (DV). A n_sample x n_y_var array where n_sample is
+        the number of observations and n_y_var is the number of DV.
+
+    exog : array-like
+        Independent variables (IV). A n_sample x n_x_var array where n is the
+        number of observations and n_x_var is the number of IV. An intercept is
+        not included by default and should be added by the user (models
+        specified using a formula include an intercept by default)
+
+    Attributes
+    -----------
+    df_resid : float
+        The number of observation `n` minus the number of IV `q`.
+    sscpr : array
+        Sums of squares and cross-products of residuals
+    endog : array
+        See Parameters.
+    exog : array
+        See Parameters.
+    design_info : patsy.DesignInfo
+        Contain design info for the independent variables if model is
+        constructed using `from_formula`
+
+    """
+    def __init__(self, endog, exog, method='svd',
+                 design_info=None, **kwargs):
+        self.design_info = design_info
+        self.method = method
+        super(_MultivariateOLS, self).__init__(endog, exog)
+
+    @classmethod
+    def from_formula(cls, formula, data, method='manova-qr', subset=None,
+                     drop_cols=None, *args, **kwargs):
+        mod = super(_MultivariateOLS, cls).from_formula(formula, data,
+                                              method=method,
+                                              subset=subset,
+                                              drop_cols=drop_cols,
+                                              *args, **kwargs)
+        return mod
+
+    def fit(self):
+        self.fittedmod = _multivariate_ols_fit(
+            self.exog, self.endog, method=self.method)
+        return _MultivariateOLSResults(self)
+
+
+class _MultivariateOLSResults(object):
+    """
+    _MultivariateOLS results class
+
+    Can be accessed as a list, each element containing a tuple (name, df) where
+    `name` is the effect (i.e. term in model) name and `df` is a DataFrame
+    containing the test statistics
+
+    """
+    def __init__(self, fitted_mv_ols):
+        self.design_info = fitted_mv_ols.design_info
+        self.exog_names = fitted_mv_ols.exog_names
+        self.endog_names = fitted_mv_ols.endog_names
+        self.fittedmod = fitted_mv_ols.fittedmod
+
+    def __str__(self):
+        return self.summary().__str__()
+
+    def __getitem__(self, item):
+        return self.results[item]
+
+    def f_test(self, hypothesis=None):
+        """
+        Testing the linear hypothesis
+            L * params * M = 0
+        where `params` is the regression coefficient matrix for the
+        linear model y = x * params, `M` is a dependent variable transform
+        matrix.
+
+        Parameters
+        ----------
+        hypothesis: A list of array-like
```

Comment:
copy changes from my previous comments.

If the entire docstring is identical, then it could be just assigned  `mv_test.__doc__ = ....__doc__`
we do this in a few places in the existing models, AFAIR.